File: sdks/dotnet/sulfone-helium/Api/Processor/Lambda.cs
using sulfone_helium.Domain.Core;
using sulfone_helium.Domain.Core.FileSystem;
using sulfone_helium.Domain.Processor;

namespace sulfone_helium.Api.Processor;

public class LambdaProcessor(Func<CyanProcessorInput, CyanFileHelper, Task<ProcessorOutput>> f)
    : ICyanProcessor
{
    public Task<ProcessorOutput> Process(CyanProcessorInput i, CyanFileHelper fileHelper)
    {
        return f(i, fileHelper);
    }
}

File: sdks/node/src/api/processor/lambda.ts
import type { CyanProcessorInput } from "../../domain/core/cyan_script_model.js";
import type { CyanFileHelper } from "../../domain/core/fs/cyan_fs_helper.js";
import type { ProcessorOutput } from "../../domain/processor/output.js";
import type { ICyanProcessor } from "../../domain/core/cyan_script.js";

type LambdaProcessorFn = (
  i: CyanProcessorInput,
  fileHelper: CyanFileHelper,
) => Promise<ProcessorOutput>;

class LambdaProcessor implements ICyanProcessor {
  private readonly _f: LambdaProcessorFn;

  constructor(f: LambdaProcessorFn) {
    this._f = f;
  }

  process(
    i: CyanProcessorInput,
    fileHelper: CyanFileHelper,
  ): Promise<ProcessorOutput> {
    return this._f(i, fileHelper);
  }
}

export type { LambdaProcessorFn };
export { LambdaProcessor };
